blog/domain-components.ftd

```diff
@@ -0,0 +1,293 @@
+-- import: fastn.com/blog/authors
+-- import: fastn-community.github.io/doc-site/common
+
+-- common.post-meta meta: Create domain-driven documentation for structured data and versatility
+published-on: November 24, 2023
+post-url: /blog/domain-components/
+author: $authors.nandini
+
+Are you someone who often grapples with the repetition of content? 
+Struggling to maintain a consistent design across various pages or 
+sections on websites? Tired of copy-pasting content and mismatched designs? 
+
+Let's dive into a relatable scenario: Imagine you want to showcase 
+customer testimonials — the fundamental part of any business site. 
+Each testimonial requires a name, a title, image and an impactful 
+quote. 
+
+However, maintaining uniformity across these testimonials is 
+a challenge. Plus, if you're working on multiple pages, duplicating 
+them becomes tedious because modifying one data turns into an 
+avalanche of updates across all the pages. The Solution...
+
+-- ds.blog-page: 
+meta: $meta
+
+-- ds.h1: Domain driven documentation in fastn
+
+With components created at domain level, you get to build content-specific pieces, 
+keeping all the unique stuff about each content type neatly packed together. 
+
+For example, you can create a testimonial card component at the domain level, 
+consolidating the structured data required for testimonials. It ensures 
+that every testimonial instance maintains a consistent layout and 
+content format, erasing the headache of replicating or managing individual 
+testimonial sections across numerous pages. 
+
+Consider the below testimonial card component and its code:
+
+-- testimonial list testimonials:
+
+-- testimonial: Nancy Bayers
```

Comment:
This record is not defined in this file. Import the file where this `testimonial` record is defined.

```suggestion
-- testimonial: Nancy Bayers
```

Reply:
@Heulitig please review the code and help me deploy this page